testslide/mock_callable.py

```diff
@@ -659,16 +666,14 @@ def and_assert_called_exactly(self, count):
         If assertion is for 0 calls, any received call will raise
         UnexpectedCallReceived and also an AssertionError.
         """
-        if not count:
-            self._runner = None
```

Comment:
We can't simply drop the existing runner, as it takes care of the accepted arguments. _Runner already implements the upper limit on calls, we don't need to repeat ourselves here, just ensure there's a runner if there's none.